src/Raven.Server/Documents/Indexes/Static/Counters/MapCountersIndex.cs

```diff
@@ -0,0 +1,248 @@
+using Raven.Server.Documents.Queries;
+using Raven.Server.Documents.Queries.Results;
+using Raven.Server.Documents.Queries.Timings;
+using Raven.Server.ServerWide.Context;
+using Voron;
+
+namespace Raven.Server.Documents.Indexes.Static.Counters
+{
+    public class MapCountersIndex : MapIndexBase<MapIndexDefinition, IndexField>
+    {
+        private readonly HashSet<string> _referencedCollections = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+
+        protected internal readonly StaticCountersIndexBase _compiled;
+        private bool? _isSideBySide;
+
+        private HandleCountersReferences _handleReferences;
+        private /*HandleCompareExchangeTimeSeriesReferences*/ object _handleCompareExchangeReferences;
+
+        protected MapCountersIndex(MapIndexDefinition definition, StaticCountersIndexBase compiled)
+            : base(definition.IndexDefinition.Type, definition.IndexDefinition.SourceType, definition)
+        {
+            _compiled = compiled;
+
+            if (_compiled.ReferencedCollections == null)
+                return;
+
+            foreach (var collection in _compiled.ReferencedCollections)
+            {
+                foreach (var referencedCollection in collection.Value)
+                    _referencedCollections.Add(referencedCollection.Name);
+            }
+        }
+
+        public override IQueryResultRetriever GetQueryResultRetriever(IndexQueryServerSide query, QueryTimingsScope queryTimings, DocumentsOperationContext documentsContext, FieldsToFetch fieldsToFetch, IncludeDocumentsCommand includeDocumentsCommand)
+        {
+            throw new NotImplementedException("TODO ppekrol");
+            //return new TimeSeriesQueryResultRetriever(DocumentDatabase, query, queryTimings, DocumentDatabase.DocumentsStorage, documentsContext, fieldsToFetch, includeDocumentsCommand);
+        }
+
+        protected override void SubscribeToChanges(DocumentDatabase documentDatabase)
+        {
+            base.SubscribeToChanges(documentDatabase);
+
+            if (documentDatabase != null)
+                documentDatabase.Changes.OnCounterChange += HandleCounterChange;
+        }
+
+        protected override void UnsubscribeFromChanges(DocumentDatabase documentDatabase)
+        {
+            base.UnsubscribeFromChanges(documentDatabase);
+
+            if (documentDatabase != null)
+                documentDatabase.Changes.OnCounterChange -= HandleCounterChange;
+        }
+
+        protected override void HandleDocumentChange(DocumentChange change)
+        {
+            if (change.Type == DocumentChangeTypes.Delete && (HandleAllDocs || Collections.Contains(change.CollectionName)))
+            {
+                // in counters we need to subscribe only to deletions of source documents
+
+                _mre.Set();
+                return;
+            }
+
+            if (_referencedCollections.Count > 0 && _referencedCollections.Contains(change.CollectionName))
```

Comment:
```suggestion
            if ( _referencedCollections.Contains(change.CollectionName))
```

Do the same thing, no?

Reply:
Yeah.